src/main.rs

```diff
@@ -487,22 +575,40 @@ fn main() {
     }
 
     let suggested_size = BloomFilter::suggest_size_in_bytes(args.expected_ngram_count);
+    println!("Suggested size is {} | Actual size is {} ", 
+             human_bytes(suggested_size as f64), human_bytes(bloom_filter.size_in_bytes() as f64));
     if suggested_size * 2 < bloom_filter.size_in_bytes() {
         println!(
             "WARNING: Your bloom filter is more than twice as large as suggested for {} elements. \
             This is good for accuracy, but it is much slower, and likely not worth the trade-off.",
             args.expected_ngram_count
         );
     }
+    // Build Progress bar (do some hacky arc/mutex wrapping)
+    let num_files = inputs.len() as u64;
+
+    let pbar = ProgressBar::new(num_files)
+        .with_style(
+            ProgressStyle::with_template(
+                "Files {human_pos}/{human_len} [{elapsed_precise}/{duration_precise}] [{wide_bar:.cyan/blue}]",
+            ).unwrap()
+        );
+    pbar.inc(0);
+    //let pbar = ProgressBar::new(num_files);
+    let now = Instant::now();
+    //pbar.set_style(ProgressStyle::with_template(
+    //    "[{elapsed_precise}] {wide_bar:0.cyan/blue} [{pos:>7}/{len:7} {eta}]").unwrap());
+    let pbar = Arc::new(Mutex::new(pbar));
 
     let threadpool = ThreadPool::new(threads);
-    for input in args.inputs {
+    for input in inputs {
         let mut output = args.output_directory.clone();
         output.push(input.file_name().unwrap());
         let bloom_filter = bloom_filter.clone();
+        let pbar = pbar.clone();
 
         threadpool.execute(move || {
-            println!("Processing {input:?}...");
+            //println!("Processing {input:?}...");
```

Comment:
Leftovers?

Reply:
Yep, should actually probably delete this. Having the println! will cause the progress bar to get re-printed for every new file